ui/components/multichain/network-list-menu/network-list-menu.js

```diff
@@ -307,6 +310,12 @@ export const NetworkListMenu = ({ onClose }) => {
             } else {
               dispatch(setActiveNetwork(network.id));
             }
+            // If presently on a dapp, communicate a change to
+            // the dapp via silent switchEthereumChain that the
+            // network has changed due to user action
+            if (useRequestQueue && selectedTabOrigin) {
+              setNetworkClientIdForDomain(selectedTabOrigin, network.id);
+            }
```

Comment:
Will this PR need a rebase to avoid regression for this?

Reply:
yes a rebase is needed after the merge of the other PR